src/router/storeSync/PositionUrlParamConfig.class.js

```diff
@@ -0,0 +1,229 @@
+import AbstractParamConfig from '@/router/storeSync/abstractParamConfig.class'
+
+function dispatchCameraPositionFromUrlIntoStore(store, urlParamValue) {
+    const allValues = urlParamValue.split(',')
+    const [x, y, z, yaw, pitch, roll] = allValues.map((value) =>
+        value === '' ? 0 : parseFloat(value)
+    )
+    store.dispatch('setCameraPosition', {
+        x,
+        y,
+        z,
+        yaw,
+        pitch,
+        roll,
+    })
+}
+
+function onlyShowIfNotZero(value) {
+    if (value === 0) {
+        return ''
+    }
+    return `${value}`
+}
+
+function generateCameraUrlParamFromStoreValues(store) {
+    if (store.state.ui.showIn3d) {
+        let { x, y, z, yaw, pitch, roll } = store.state.position.camera
+        return [x, y, z, yaw, pitch, roll].map((value) => onlyShowIfNotZero(value)).join(',')
+    }
+    return null
+}
+
+function readValueFromObjectOrReturnNull(object, paramName, type) {
+    if (object && paramName in object) {
+        return type(object[paramName])
+    }
+    return null
+}
+
+function parseCameraFromQuery(query) {
+    const camera = readValueFromObjectOrReturnNull(query, 'camera', String)
+    if (camera) {
+        let cameraValues = camera.split(',')
+        // the split must have 6 components (x, y, z, pitch, yaw and roll)
+        if (cameraValues.length === 6) {
+            // parsing to number all values (default to 0 if the value is empty)
+            cameraValues = cameraValues.map((value) => (value === '' ? 0 : Number(value)))
+            const [x, y, z, pitch, yaw, roll] = cameraValues
+            return {
+                x,
+                y,
+                z,
+                pitch,
+                yaw,
+                roll,
+            }
+        }
+    }
+    return null
+}
+
+function populateStoreWithParam(store, query, paramName, storeValue, dispatchTo) {
+    if (query && query.hasOwnProperty(paramName)) {
+        const value = Number(query[paramName])
+        if (value !== storeValue) {
+            return store.dispatch(dispatchTo, value)
+        }
+    }
+}
+
+/**
+ * Definition of a set of URL params to store the position of the map in the URL. These params will
+ * change depending on the mode of rendering of the map (2D or 3D)
+ *
+ * In 2D it will generate lon/lat/z params.
+ *
+ * In 3D it will generate a unique camera URL param that will be a concat of all relevant camera
+ * values (x,y,z,yaw,pitch,roll)
+ *
+ * This param config class takes for granted that the store's flag showIn3d has been processed
+ * before it is then called to do its job. Meaning that if the flag has changed in the URL, but the
```

Comment:
How can this be guarantied that the showIn3d gets processed first ?

Reply:
by adding AFTER the param parser for the 3D flag, see `src/router/storeSync/storeSync.config.js` line 31 and further